### DISCUSSION TITLE
Download error with curl: (6) getaddrinfo() thread failed to start and the workaround

### DISCUSSION THREAD

**Discussion by AnyOfYou** [itzg/docker-minecraft-bedrock-server#361] [General]
Hi, I have a old linux server(Ubuntu 16.04 with glibc 2.23) running bedrock server with this project for years. Today when i try to pull the latest image and update the server version, it ends quickly with this error 'Download error with curl: (6) getaddrinfo() thread failed to start', the debug log is down below.

<details><summary>Details</summary>
<p>

++ pwd
+ echo '       current directory is /data'
+ downloadPage=https://www.minecraft.net/en-us/download/server/bedrock
+ [[ TRUE != TRUE ]]
+ case ${VERSION^^} in
+ echo 'Looking up latest version...'
+ lookupVersion serverBedrockLinux
+ platform=serverBedrockLinux
+ for i in {1..3}
       current directory is /data
Looking up latest version...
++ restify --user-agent=itzg/minecraft-bedrock-server --headers 'accept-language:*' --attribute=data-platform=serverBedrockLinux https://www.minecraft.net/en-us/download/server/bedrock
++ jq -r '.[0].href'
+ DOWNLOAD_URL=https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip
+ [[ -n https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip ]]
+ break 2
+ [[ -z https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip ]]
+ [[ https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip =~ http.*/.*-(.*)\.zip ]]
+ VERSION=1.20.32.03
+ rm -f restify.err
+ '[' '!' -f bedrock_server-1.20.32.03 ']'
+ [[ ! -n https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip ]]
+ [[ /tmp != /tmp ]]
++ basename https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip
+ TMP_ZIP=/tmp/bedrock-server-1.20.32.03.zip
+ echo 'Downloading Bedrock server version 1.20.32.03 ...'
+ curl -v -o /tmp/bedrock-server-1.20.32.03.zip -fsSL https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip
Downloading Bedrock server version 1.20.32.03 ...
* getaddrinfo() thread failed to start
* Could not resolve host: minecraft.azureedge.net
* Closing connection 0
curl: (6) getaddrinfo() thread failed to start
+ echo 'ERROR failed to download from https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.32.03.zip'
+ echo '      Double check that the given VERSION is valid'
+ exit 2

</p>
</details> 

Its weird because the previous step (use restify to get latest server version) is successful, and i could download the zip file no problem with curl command on my server manually, the network of my server is okay.

So I go dig more about this issue, and found some related posts, u could read more at [here](https://openforums.blog/2021/12/15/tracing-curl-error-6-getaddrinfo-thread-failed-to-start-error-with-systemd-seccomp-filters/) and [here](https://github.com/adoptium/containers/issues/215#issuecomment-1142046045). Although the details may have some differents, but I believe this issue is related with system's glibc version/docker/curl.

My temporary fix is add "--security-opt seccomp=unconfined" option with docker run command, its succeed, so I shard this finding to help.

**Comment by itzg** [comment]
Thanks for sharing. 

I’m going to convert this to a discussion so it doesn’t get closed out.

**Comment by zpertee** [comment]
Thanks @AnyOfYou!  This was causing me an issue this morning and sure enough your temporary fix got me going again as well.

**Comment by huapox** [comment]
+1
--
ok with: `docker run -it --rm --security-opt seccomp=unconfined 172.25.20.115/base/env-system:openeuler22-v1` or just

`docker run -it --rm --privileged 172.25.20.115/base/env-system:openeuler22-v1`

**Reply by n-busireddy** [reply]
Will this work with the docker build build command as well?
